### PR TITLE
fire-hpp: add version 1.0

### PR DESCRIPTION
**Pull request by toge**
### Summary
Changes to recipe:  **fire-hpp/1.0**

#### Motivation
This is the first stable release.

#### Details
https://github.com/kongaskristjan/fire-hpp/compare/v0.2...v1.0

---
- [x] Read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md)
- [x] Checked that this PR is not a duplicate: [list of PRs by recipe](https://github.com/conan-io/conan-center-index/discussions/24240)
- [x] Tested locally with at least one configuration using a recent version of Conan

### CODE REVIEW

File: recipes/fire-hpp/all/conandata.yml [hunk 1]
```diff
@@ -1,4 +1,7 @@
 sources:
+  "1.0":
+    url: "https://github.com/kongaskristjan/fire-hpp/archive/v0.2.tar.gz"
```
Comment (ErniGH) [inline recipes/fire-hpp/all/conandata.yml:3]: Hey @toge, the URL is from the previous version

Comment (toge) [reply recipes/fire-hpp/all/conandata.yml]: @ErniGH 
Sorry, I fixed.

**Review by ErniGH — APPROVED** [review]
LGTM! Thanks

**Review by AbrilRBS — APPROVED** [review]
Thanks @toge!

I see in https://github.com/kongaskristjan/fire-hpp/blob/master/docs/conan.md that upstream expects users to use their own conanfile. Maybe we could opena. PR to also mention the usual `conan install --requires=fire-hpp/1.0` approach, what do you think? Are you up for it? :)

**Pull request merged**